packages/db/src/models/base.ts

```diff
@@ -0,0 +1,58 @@
+import Knex from 'knex';
+
+import { RecordDoesNotExist } from '@cumulus/errors';
+
+import { tableNames } from '../tables';
+
+import { isRecordDefined } from '../database';
+
+class BasePgModel<ItemType, RecordType extends { cumulus_id: number }> {
+  readonly tableName: tableNames;
+
+  constructor({
+    tableName,
+  }: {
+    tableName: tableNames,
+  }) {
+    this.tableName = tableName;
+  }
+
+  get(
+    knexOrTrx: Knex | Knex.Transaction,
+    params: Partial<RecordType>
+  ) {
+    return knexOrTrx<RecordType>(this.tableName).where(params).first();
+  }
+
+  async getRecordCumulusId(
+    knexOrTrx: Knex|Knex.Transaction,
```

Comment:
At least based on how the calling code is currently written, I'm not sure that the proposed refactor would work, because we only have the transaction reference in scope, not the full knex client. For example:

https://github.com/nasa/cumulus/blob/feature-rds/packages/api/lambdas/sf-event-sqs-to-db-records/write-pdr.js#L70

However, I think this could work:

```
async getRecordCumulusId( param: {
    knex: Knex | undefined,
    whereClause : Partial<RecordType>,
    transaction: Knex.Transaction | undefined
}): Promise<number> {
   if (!knex && !transaction) {
     throw new Error('A Knex client or transaction is required')
   }
   const client = knex ? knex(this.tableName) : trx(this.tableName);
   const record: Recordtype = await client
    .select('cumulus_id')
    .where(whereClause);
   ...
}
```

What do you think?

We're using this `knexOrTransaction` throughout the models, so if we're going to make this change it should be across the board

Reply:
As I re-examine my proposed refactor, I don't think it adds anything other than validating that you are passing in a full knex client or a transaction.

Reply:
Ensure that we're testing for both a full knex client and transaction

Reply:
5bc66c36da01f0b57dc1b7befb6134ab702e13e3